test/VaultRegistry.test.ts

```diff
@@ -1,445 +1,943 @@
+                                this.newProtocolGovernance.address
+                            );
+                        await sleep(this.governanceDelay);
+                        await this.subject
+                            .connect(this.admin)
+                            .commitStagedProtocolGovernance();
+                        expect(
+                            await this.subject.stagedProtocolGovernanceTimestamp()
+                        ).to.be.equal(0);
+                    });
+                });
+            });
+        });
+
+        describe("#vaultsCount", () => {
+            it("returns the number of registered vaults", async () => {
+                expect(await this.subject.vaultsCount()).to.be.equal(1);
+            });
+
+            describe("access control:", () => {
+                it("allowed: any address", async () => {
+                    await withSigner(randomAddress(), async (s) => {
+                        await expect(this.subject.connect(s).vaultsCount()).to
+                            .not.be.reverted;
+                    });
+                });
+            });
+
+            describe("properties", () => {
+                pit(
+                    "@property: when N new vaults have been registered, vaults count will be increased by N",
+                    { numRuns: RUNS.verylow },
+                    integer({ min: 1, max: 5 }),
+                    async (vaultsCount: number): Promise<boolean> => {
+                        let oldVaultsCount = Number(
+                            await this.subject.vaultsCount()
+                        );
+                        for (var i = 0; i < vaultsCount; ++i) {
+                            await deployments.deploy("MockERC165", {
+                                from: this.deployer.address,
+                                contract: "MockERC165",
+                                args: [],
+                                autoMine: true,
+                            });
+                            let newMock = await ethers.getContract(
+                                "MockERC165"
+                            );
+                            await newMock.allowInterfaceId(VAULT_INTERFACE_ID);
+
+                            this.subject
+                                .connect(this.allowedRegisterVaultSigner)
+                                .registerVault(
+                                    newMock.address,
+                                    this.ownerSigner.address
+                                );
+                        }
+                        let newVaultsCount = Number(
+                            await this.subject.vaultsCount()
+                        );
+                        expect(newVaultsCount - oldVaultsCount).to.be.equal(
+                            vaultsCount
+                        );
+                        expect(
+                            (await this.subject.vaults()).length
```

Comment:
pls check that if you use (await this.subject.vaults()).length + 1 here - the test will fail